client/src/services/casperServices.js

```diff
@@ -35,18 +36,20 @@ export const buildContractInstallDeploy = (baseAccount, session) => {
  * @param {Deploy} deploy main account public key
  * @param {String} mainAccountHex hash contract content
  * @param {String} setAccountHex contract's arguments
+ * @param {Object} ledgerOptions ledger's options
  * @returns {Deploy} Signed deploy
  */
-export const signDeploy = async (deploy, mainAccountHex, setAccountHex, casperApp) => {
+export const signDeploy = async (deploy, mainAccountHex, setAccountHex, { casperApp = null, keyPath = 0 }) => {
 	try {
 		if (!casperApp) {
 			const deployObj = DeployUtil.deployToJson(deploy);
 			const signedDeploy = await Signer.sign(deployObj, mainAccountHex, setAccountHex);
 			return signedDeploy;
 		} else {
+			toast('Transaction submitted. Awaiting your approval in the ledger.');
```

Comment:
Pls handle it in React component, need to isolate UI handling with services.

Reply:
I use this way because I cannot find the way to return the signal that ledger is pending for review. The casper ledger lib doesn't have any API to check it.

Reply:
You can toast the message right after user click on confirm button, or update the loading status is enough.

Reply:
Yeah! I will toast messages after user clicks button then. Plz have a look here https://github.com/CasperDash/casperdash-client/pull/69/files#diff-662b434baf645b2ddf62bbc19882f340805c5bee803a3955cdfec0f8ea88c0e1R57